cli/internal/typer/orchestrator.go

```diff
@@ -0,0 +1,150 @@
+package typer
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/rudderlabs/rudder-iac/cli/internal/app"
+	"github.com/rudderlabs/rudder-iac/cli/internal/logger"
+	"github.com/rudderlabs/rudder-iac/cli/internal/typer/generator/core"
+	"github.com/rudderlabs/rudder-iac/cli/internal/typer/generator/platforms/kotlin"
+	"github.com/rudderlabs/rudder-iac/cli/internal/typer/plan"
+)
+
+var (
+	orchestratorLog = logger.New("typer.orchestrator")
+)
+
+// Orchestrator coordinates the RudderTyper code generation process
+type Orchestrator struct {
+	deps app.Deps
+}
+
+// GenerationOptions contains configuration for code generation
+type GenerationOptions struct {
+	TrackingPlanID string
```

Comment:
That tracking plan id needs to be an internal detail of the PlanProvider, according to the design doc, not something baked in RudderTyper